lib/compiler.js

```diff
@@ -12,7 +12,13 @@ const AV = require("./assignment-visitor.js");
 const assignmentVisitor = new AV;
 
 const shebangRegExp = /^#!.*/;
-const importExportRegExp = /\b(?:im|ex)port\b/;
+
+// Matches any import or export identifier as long as it's not preceded by
+// a `.` character (to avoid matching module.export, for example). Because
+// Reify replaces all import and export declarations with module.import
+// and module.export calls, this logic should prevent the compiler from
+// ever having to recompile code it has already compiled.
+const importExportRegExp = /(?:^|[^.])\b(?:im|ex)port\b/;
```

Comment:
It looks like its better to do two regexes.
The original light one of  `/\b(?:im|ex)port\b/` then one for `module\.(im|ex)port`